app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java

```diff
@@ -0,0 +1,163 @@
+package com.evelynhasama.bucketbuds;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class BucketFriendsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>{
+
+    private Context context;
+    List<BucketFriendListItem> allItemsList;
+    public static final String TAG = "BucketFriendsAdapter";
+
+    public BucketFriendsAdapter(Context context, List<BucketFriendListItem> allItemsList) {
+        this.allItemsList = allItemsList;
+        this.context = context;
+    }
+
+    @NonNull
+    @NotNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
+
+        RecyclerView.ViewHolder viewHolder;
+        View view;
+
+        switch (viewType) {
+            case BucketFriendListItem.TYPE_FRIEND:
+                view = LayoutInflater.from(context).inflate(R.layout.item_add_friend, parent, false);
+                viewHolder = new FriendViewHolder(view);
+                break;
+
+            case BucketFriendListItem.TYPE_HEADER:
+                view = LayoutInflater.from(context).inflate(R.layout.item_bucket_friend_header, parent, false);
+                viewHolder = new HeaderViewHolder(view);
+                break;
+            default:
+                throw new IllegalStateException("Unexpected value: " + viewType);
+        }
+
+        return viewHolder;
```

Comment:
nit: I'm scratching my head around these `TYPE` things, can you please add a comment with a tl;dr?